drop-in/src/main/java/com/adyen/checkout/dropin/ui/LazyArguments.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 20/9/2022.
+ */
+
+package com.adyen.checkout.dropin.ui
+
+import android.app.Activity
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import kotlin.reflect.KProperty
+
+internal inline fun <reified T> Activity.arguments(key: String): LazyProvider<Fragment, T> =
+    argumentDelegate(key) { intent?.extras }
+
+internal inline fun <reified T> Fragment.arguments(key: String): LazyProvider<Fragment, T> =
+    argumentDelegate(key) { arguments }
+
+internal inline fun <A, reified T> argumentDelegate(
+    key: String,
+    crossinline provideArguments: (A) -> Bundle?
+): LazyProvider<A, T> = object : LazyProvider<A, T> {
+    override fun provideDelegate(argumentsFactory: A, prop: KProperty<*>): Lazy<T> = lazy {
```

Comment:
Although it seems there is no use, it is actually needed in order for Kotlin's property delegation to work (just take a look at the `getValue` method in the ViewBinding magic you linked)

Reply:
Can you elaborate more about the kotlin property delegation? as the link I sent it was extending `KotlinProperty` and we're passing it here as an argument.

Reply:
Take a look at this method (screenshot taken from the link as example). This `getValue` method needs the `KProperty` parameter. Instead of `ReadOnlyProperty` I used `Lazy`. It hides the implementation of `getValue`, but essentially works the same plus some extra benefits like thread safety
![image](https://user-images.githubusercontent.com/17701279/193024767-16b253d7-da34-4b38-a26e-d3b029c9cd53.png)